### ISSUE TITLE
[Bug]: When Carousel is added, "unhandledRejection" is shown in terminal on `nuxt dev`.

### ISSUE THREAD

**Issue by Jamiewarb** [radix-vue/shadcn-vue#262]
### Environment

```bash
Developement/Production OS: Macbook Pro Apple M1 Pro MacOS 13.4.1 
Node version: v18.17.1
Package manager: pnpm@8.14.0
Radix Vue version: 1.3.0
Shadcn Vue version: 0.8.6
Vue version: 3.4.6
Nuxt version: 3.9.0
Nuxt mode: universal
Nuxt target: server
CSS framework: tailwindcss@3.3.3
Client OS: Windows 10 19043.1110
Browser: Chrome 90.0.4430.212
```

### Link to minimal reproduction

https://github.com/hex-digital/shadcn-vue-test

### Steps to reproduce

- Clone minimal repo (or just follow shadcn-vue installation instructions, and then add Carousel via `pnpm dlx shadcn-vue@latest carousel`)
- `pnpm install`
- `pnpm dev`
- Observe error

### Describe the bug

Following error is displayed

```
 ERROR  [unhandledRejection] Line 8: Unexpected identifier                                                                        17:05:27

  at ErrorHandler.constructError (node_modules/.pnpm/esprima@4.0.1/node_modules/esprima/dist/esprima.js:5012:22)
  at ErrorHandler.createError (node_modules/.pnpm/esprima@4.0.1/node_modules/esprima/dist/esprima.js:5028:27)
  at Parser.unexpectedTokenError (node_modules/.pnpm/esprima@4.0.1/node_modules/esprima/dist/esprima.js:1985:39)
  at Parser.throwUnexpectedToken (node_modules/.pnpm/esprima@4.0.1/node_modules/esprima/dist/esprima.js:1995:21)
  at Parser.expect (node_modules/.pnpm/esprima@4.0.1/node_modules/esprima/dist/esprima.js:2168:19)
  at Parser.parseExportDeclaration (node_modules/.pnpm/esprima@4.0.1/node_modules/esprima/dist/esprima.js:4935:19)
  at Parser.parseStatementListItem (node_modules/.pnpm/esprima@4.0.1/node_modules/esprima/dist/esprima.js:3367:39)
  at Parser.parseModule (node_modules/.pnpm/esprima@4.0.1/node_modules/esprima/dist/esprima.js:4715:29)
  at Object.parse (node_modules/.pnpm/esprima@4.0.1/node_modules/esprima/dist/esprima.js:122:38)
  at Object.parse (node_modules/.pnpm/recast@0.23.4/node_modules/recast/parsers/esprima.js:14:34)
```

### Expected behavior

No error

### Conext & Screenshots (if applicable)

<img width="1131" alt="image" src="https://github.com/radix-vue/shadcn-vue/assets/2754728/a9785942-e610-4fd0-b641-4128156003f5">

**Comment by Jamiewarb** [comment]
Looks like Carousel was introduced for #220 and #226 and closed in #227

**Comment by sadeghbarati** [comment]
Hey @Jamiewarb you should use 

#### pnpm dlx shadcn-vue@latest **`((( add )))`** carousel

Had problem with installing carousel files with CLI but not with `esprima` module

`pnpm dlx shadcn-vue@latest add carousel` maybe related to my internet speed :x:

`pnpm dlx shadcn-vue@latest add` and choose the carousel from radio items ✅ 

![image](https://github.com/radix-vue/shadcn-vue/assets/17789047/c0ce020b-1ca2-4e3f-b5d9-342037188393)

**Comment by sadeghbarati** [comment]
your issue is related to Nuxt auto-import feature?

---


#### Manual module 

Just tested the manual `modules/shadcn.ts` way it's working well 👍 

https://stackblitz.com/edit/github-juxvdk?file=modules%2Fshadcn.ts,package.json,nuxt.config.ts


#### `shadcn-nuxt` module

shadcn-nuxt `recast` package has an error when the export type is present in index.ts

```ts
export type {
  EmblaCarouselType as CarouselApi,
} from 'embla-carousel'
```

https://stackblitz.com/github/hex-digital/shadcn-vue-test?file=nuxt.config.ts,app.vue

I think It's because we should use recast typescript parser 


---

@zernonia 

[OXC AST playground](https://oxc-project.github.io/oxc/playground/?code=3YCAAIAugYCAgICAgICyHorESipoTq%2BSwGz8oO22DnZHMs4HmY7oRY90NAHfWV45deSUhqqjAVWgUYvcSMjnyDg3t2LuA%2FHHsQKt3ULT1EefdmyXl7oygI3iSzEUuNCk73RneLXmgYjoUgbT7wOR8kVObTl4anLHO926AMl8B8l1tJzdbp3KtFQYsr%2Fw%2BtNs%2F6OXuN0cmbc0QgDoLXoi3aA%3D)

Here is my ugly trying with magicast

<details><summary>Details</summary>
<p>

```ts
import { readFileSync, readdirSync } from 'node:fs'
import { join } from 'node:path'
import { addComponent, createResolver, defineNuxtModule } from '@nuxt/kit'
import type { ExportNamedDeclaration } from '@babel/types'
import { loadFile } from 'magicast'

// Module options TypeScript interface definition
export interface ModuleOptions {
  /**
   * Prefix for all the imported component
   */
  prefix?: string
  /**
   * Directory that the component lives in.
   * @default "./components/ui"
   */
  componentDir?: string
}

export default defineNuxtModule<ModuleOptions>({
  meta: {
    name: 'shadcn',
    configKey: 'shadcn',
  },
  defaults: {
    prefix: '',
    componentDir: './components/ui',
  },
  async setup({ prefix, componentDir }, nuxt) {
    const t = nuxt.options.typescript
    const COMPONENT_DIR_PATH = componentDir!
    const ROOT_DIR_PATH = nuxt.options.rootDir
    const { resolve, resolvePath } = createResolver(ROOT_DIR_PATH)

    nuxt.hook('components:dirs', (dirs) => {
      dirs.unshift({
        path: resolve(COMPONENT_DIR_PATH),
        extensions: [],
      })
    })

    try {
      readdirSync(resolve(COMPONENT_DIR_PATH))
        .forEach(async (dir) => {
          const filePath = await resolvePath(join(COMPONENT_DIR_PATH, dir, 'index'), { extensions: ['.ts', '.js'] })

          const content = await loadFile(filePath, {
            // eslint-disable-next-line @typescript-eslint/no-require-imports
            parser: require('recast/parsers/typescript'),  ////////////////////////  How about js Vue/JS project
          })
          const program = content.$ast.type === 'Program' && content.$ast

          const test = program.body.filter((statement): statement is ExportNamedDeclaration => statement.type === 'ExportNamedDeclaration' && statement.exportKind === 'value').flatMap((node) => node.source.value)

          console.log(test)
          // const exportedKeys: string[] = ast.program.body
          //   .filter(node => node.type === 'ExportNamedDeclaration')
          //   .flatMap(node => node.specifiers.map(specifier => specifier.exported.name))
          //   .filter((key: string) => /^[A-Z]/.test(key))

          // exportedKeys.forEach((key) => {
          //   addComponent({
          //     name: `${prefix}${key}`, // name of the component to be used in vue templates
          //     export: key, // (optional) if the component is a named (rather than default) export
          //     filePath: resolve(filePath),
          //   })
          // })
        })
    }
    catch (err) {
      if (err instanceof Error)
        console.warn(err.message)
    }
  },
})
```


</p>
</details>

**Comment by zernonia** [comment]
Thanks @sadeghbarati . Decided to use `oxc-parser`. Tested with js file too and it works.  Fixed in `0.8.7`

**Comment by Jamiewarb** [comment]
Thank you for the exceptionally quick fix ❤️
